sale_order_type/models/account_invoice.py

```diff
@@ -24,6 +24,16 @@ def _onchange_partner_id(self):
     @api.onchange('sale_type_id')
     def onchange_sale_type_id(self):
         if self.sale_type_id.payment_term_id:
-            self.payment_term = self.sale_type_id.payment_term_id.id
+            self.payment_term_id = self.sale_type_id.payment_term_id.id
```

Comment:
Good catch!

Reply:
That is thanks to https://github.com/OCA/sale-workflow/pull/716#pullrequestreview-175395682